packages/trader/src/Stores/Modules/Trading/Constants/__tests__/validation-rules.spec.ts

```diff
@@ -0,0 +1,163 @@
+                {
+                    barrier_1: '123',
+                    barrier_2: '123',
+                }
+            )
+        ).toBe(true);
+    });
+
+    it('should contain rules for barrier_2', () => {
+        mocked_store.barrier_count = 2;
+
+        expect(returned_validation_rules).toHaveProperty('barrier_2');
+        expect(returned_validation_rules.barrier_2.rules[0][1].condition(mocked_store)).toBe(true);
+        expect(returned_validation_rules.barrier_2.rules[1][1].condition(mocked_store)).toBe(2);
+        expect(returned_validation_rules.barrier_2.rules[2][1].func(...default_mocked_params)).toBe(false);
+        expect(returned_validation_rules.barrier_2.rules[3][1].func(...default_mocked_params)).toBe(false);
+    });
+
+    it('should contain rules for duration', () => {
+        expect(returned_validation_rules).toHaveProperty('duration');
+        expect(returned_validation_rules.duration.rules[0][1].message).toBe('Duration is a required field.');
+    });
+
+    it('should contain rules for start_date', () => {
+        expect(returned_validation_rules).toHaveProperty('start_date');
+        expect(returned_validation_rules.start_date.trigger).toBe('start_time');
+    });
+
+    it('should contain rules for expiry_date', () => {
+        expect(returned_validation_rules).toHaveProperty('expiry_date');
+        expect(returned_validation_rules.expiry_date.trigger).toBe('expiry_time');
+    });
+
+    it('should contain rules for start_time', () => {
+        expect(returned_validation_rules).toHaveProperty('start_time');
+        expect(returned_validation_rules.start_time.rules[0][1].func(...start_time_mocked_params)).toBe(false);
+        expect(returned_validation_rules.start_time.rules[1][1].func(...start_time_mocked_params)).toBe(false);
+        expect(returned_validation_rules.start_time.rules[2][1].func(...start_time_mocked_params)).toBe(false);
+
+        mocked_store.contract_start_type = 'spot';
+        expect(returned_validation_rules.start_time.rules[3][1].func(...start_time_mocked_params)).toBe(true);
+
+        mocked_store.contract_start_type = 'test_spot';
+        expect(returned_validation_rules.start_time.rules[3][1].func(...start_time_mocked_params)).toBe(false);
+
+        (isTimeValid as jest.Mock).mockReturnValueOnce(true);
+        (isSessionAvailable as jest.Mock).mockReturnValueOnce(true);
+        expect(returned_validation_rules.start_time.rules[3][1].func(...start_time_mocked_params)).toBe(true);
+    });
+
+    it('should contain rules for expiry_time', () => {
+        expect(returned_validation_rules).toHaveProperty('expiry_time');
+        expect(returned_validation_rules.expiry_time.rules[0][1].func(...expiry_time_mocked_params)).toBe(false);
+        expect(returned_validation_rules.expiry_time.rules[1][1].func(...expiry_time_mocked_params)).toBe(false);
+        expect(returned_validation_rules.expiry_time.rules[2][1].func(...expiry_time_mocked_params)).toBe(false);
```

Comment:
```suggestion
        expect(returned_validation_rules.expiry_time.rules[2][1].func(...expiry_time_mocked_params)).toBeFalsy();
```
nit-pick :)

Reply:
Got a question: what are the benefits of using `.toBeFalsy()` instead of `.toBe(false)`? Asking because I wrote `.toBeFalsy()` and then replaced it with `.toBe(false)` :)
Tested functions returned boolean so I checked exact value, not falsy or truthy

Reply:
Like to make it safer in case if some function return undefined instead of false?